### ISSUE TITLE
aria-details violation raised by Ace/Axe

### ISSUE THREAD

**Issue by marisademeglio** [daisy/epub-accessibility-tests#48]
In both Math and Extended Descriptions books, Ace raises violations (via Axe) about `aria-details`. 

I believe we are using it correctly, according to this:
https://www.w3.org/TR/wai-aria-1.1/#aria-details

The problem can be reproduced here by running Axe (via Firefox plugin) on this file:
https://marisademeglio.github.io/temp/aria-details-test.html

**Comment by rdeltour** [comment]
`aria-details` is currently not supported in Axe (see [axe-core v3.4.0 doc on ARIA support](https://github.com/dequelabs/axe-core/blob/v3.4.0/doc/aria-supported.md). We could override this in Ace (by patching Axe or with a custom rule). My understanding was that the test books would be used to see if the technique had sufficient support and was viable as a recommendation; that's why we didn't update Ace yet.

**Comment by marisademeglio** [comment]
In the case of my example document (linked above), the information provided by `@aria-details` is an enhancement over what’s in the `@alt` attribute. But an aXe check gives a better result if I just remove `@aria-details` completely. As in, because I’ve used a not-widely-supported accessibility feature, my document is considered less accessible than if I left it out completely. I get that less aria can be the best aria, but this isn’t quite that case. 

This doesn’t encourage proper use of the standard - I think it actually discourages authors from following the W3C aria spec example.

What's the minimum threshold for sufficient support for accessibility features? As of (this blog post)[https://www.scottohara.me/blog/2018/11/08/aria-details.html] (a year old), it doesn't look good for `aria-details`, with only two AT/browser combos giving some sort of support. Then again, only 8 were tested. 

It will be interesting to see if reading systems testing produces any instances of additional support; however, given how close they are technologically, I feel like it will just mimic the state of AT/browser combo results.

**Comment by rdeltour** [comment]
Axe’s [“How we make decision on rules”](https://github.com/dequelabs/axe-core/blob/v3.4.0/doc/accessibility-supported.md#impact-on-aria) documentation is particularly interesting here. Here’s a copy of their decision tree for ARIA support:

> 1. If the feature's use is supported by all platforms: allow, else
> 2. If the feature's use does not have a negative impact on accessibility: allow, else
> 3. If we can detect a fallback: allow, else
> 4. disallow the feature's use until it is supported

I agree that Axe’s rejection of `aria-details` is probably too strong. I don't think that the feature can have a negative impact on accessibility, so it should be allowed as per (2); or at least as per (3) if the details content is both visible and following the described element in document order.

**Comment by rdeltour** [comment]
Reported to Axe folks as dequelabs/axe-core#1931

**Comment by GeorgeKerscher** [comment]
Hi,

 

While not fully supported, there is partial  useful support. JAWS does report that details are present.

 

Best

George

**Comment by danielweck** [comment]
For what this is worth: https://a11ysupport.io/tech/aria/aria-details_attribute

**Comment by ways2read** [comment]
We decided to close this in the meeting on 17 October 2023 since this is an Ace issue and marked as fixed by the folk at Deque.
